### ISSUE TITLE
Incorrect tax detail in invoice when a cart rule with a gift is applied

### ISSUE THREAD

**Issue by idnovate** [PrestaShop/PrestaShop#34085]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

If you apply a cart rule with a gift to an order, and the order has products with different tax rate, the tax detail is not calculated correctly.

### Steps to reproduce

1. Configure 2 products with different tax rate
![image](https://github.com/PrestaShop/PrestaShop/assets/15104724/c75c6e8f-b4d4-470a-a202-3f8c85202cd1)

![image](https://github.com/PrestaShop/PrestaShop/assets/15104724/4d78e7aa-0318-451c-b9f2-8acb1ec1eecf)

2. Create a cart rule with one of these products as a gift
![image](https://github.com/PrestaShop/PrestaShop/assets/15104724/009c8d88-47da-436d-be25-6e3fe92b1ad8)

3. Make an order and change the status to generate the invoice

4. Check the tax detail in the invoice. You have only paid for one of these products, but the tax detail is prorated between the 2 products
![image](https://github.com/PrestaShop/PrestaShop/assets/15104724/8891424e-3a66-4e33-8d7c-e07bdd4ffa9e)

### PrestaShop version(s) where the bug happened

8.1.1

### Your company or customer's name goes here (if applicable).

idnovate.com

**Comment by sallemiines** [comment]
Hello @idnovate 

This issue is duplicate [#17760](https://github.com/PrestaShop/PrestaShop/issues/17760)

Thank you

**Comment by sallemiines** [comment]
duplicate of [#17760 ](https://github.com/PrestaShop/PrestaShop/issues/17760)

**Comment by idnovate** [comment]
I think it's not related with https://github.com/PrestaShop/PrestaShop/issues/17760.

This issue happens when there are products with different TAX rate and you use a cart rule to gift one of them. The TAX detail is not computed correctly.

The other issue should be fixed from the "Gift card" modules, not from PrestaShop. The problem is that the TAX rate should be applied when the gift card is used, and it isn't because it's managed as a cart rule, not as a payment method.

**Comment by AureRita** [comment]
Hi @idnovate 

Thank your for your details, 
Currently your issue seems to be the same than your previous issue : https://github.com/PrestaShop/PrestaShop/issues/33543 this one will be corrected on 8.1.2 not on 8.1.1. 

Another issue seems to be the same : #33912 

Are you agree with me ot not ? 

Waiting for your feedback

**Comment by idnovate** [comment]
They are not related, this is a different issue :)

**Comment by AureRita** [comment]
Hi @idnovate ,

Thank you for your report and your patience, I think I reproduce the issue with PrestaShop version 8.1.2, as you can see : 

[Untitled_ Oct 27, 2023 3_26 PM.webm](https://github.com/PrestaShop/PrestaShop/assets/100699445/adb73b23-4f5e-4ccb-883a-71d0cfb39366)

I'll add this to the backlog so it can be fixed.

Please be aware that some issues might take a very long time to be resolved.

Thanks

**Comment by idnovate** [comment]
Thanks! I already fixed it, but the tests are missing :)

https://github.com/PrestaShop/PrestaShop/pull/34086

**Comment by florine2623** [comment]
Hello @ibahloul-ps ,

Can this topic be checked ?

Thanks!

**Comment by boo-code** [comment]
@florine2623 — checked, and the short answer is that the fix in #34086 is still coherent against current code but its test does not run, so nothing is actually verifying it.

`Order::getProductTaxesDetails()` on `9.2.x` is unchanged in the region the PR touches: the `getCartRules()` loop still handles only `reduction_product`, and `$order_details` is still fetched after it. So the change still makes sense two years on.

The scenario added to the PR in September 2025 is the problem. Putting its feature file onto a current checkout and asking behat to resolve it:

```
1 scenario (1 undefined)
25 steps (12 undefined, 13 skipped)
```

Twelve of its twenty-five steps have no definition — `the cart total excluding tax should be`, `I generate an invoice for order`, `the order invoice VAT breakdown for order … should be:` and others were written without the step definitions to back them. Steps used by the eighteen scenarios already in that same file resolve normally, so this is specific to the new scenario rather than a problem with my setup. A dry run reports it undefined; a real run skips it. Details are on the PR.

So this issue is not waiting on a diagnosis — the mechanism is understood and a candidate fix exists — it is waiting on that scenario being expressed with step definitions that exist, after which the numbers can actually be checked.
